doc/VERSION.md

```diff
@@ -59,24 +96,34 @@ a complex topic, so better don't underestimate it, but
 generally said the methods should behave on the same data
 the same way.
 
+As we use [symbol versioning](dev/symbol-versioning.md), the SO_VERSION
+of Elektra always remains the same, even if the major version changes.
+That means, if we rename or remove a function in the `2.0` release,
+applications that linked against Elektra `1.0` can still link against
+Elektra `2.0` but they do not necessarily compile with Elektra `2.0`
+anymore. Once the code is adapted to use `2.0`, it cannot link against
+`1.0` anymore. Note that this feature is only available on platforms
+that support symbol versioning. For other platforms, all applications
+need to be recompiled for every major version.
+
 References:
 http://tldp.org/HOWTO/Program-Library-HOWTO/shared-libraries.html
 http://packages.debian.org/de/sid/icheck
 
-## Increment
-
-This section describes how to increment the `KDB_VERSION`.
-It consists of a triplet integer `current:revision:age`.
-
-`revision` is something which will always incremented when there
-is a new bug fix release.
-
-`current` and `age` will be incremented by one when you release
-a compatible but changed API. The revision is set back to zero then.
-
-Note: All 3 versioning infos are handled separately!
-
-http://www.gnu.org/software/libtool/manual/libtool.html#Versioning
-http://semver.org/
-
-> TODO write about SO_VERSION and rest version
+## Bindings
+
+Bindings are in general tied to Elektra's version number except
```

Comment:
Semantic versioning allows to add a build metadata suffix. We could use that for the binding version, e.g. `1.2.3+rust1` is the first version of the Rust binding for `1.2.3`, `1.2.3+rust2` is a second version that fixes some bugs only in the binding, etc.

Otherwise, I would clarify that the patch version is essentially language specific. That this is the case is easy to see:
1. `1.0.0` of the C API is released
2. `1.0.0` of the Rust API is released at the same time
3. `1.0.1` of the Rust API is released with a bugfix
4. `1.0.2` of the Rust API is released with another bugfix
5. `1.0.1` of the C API is released
6. `1.0.3` of the Rust API is released to incorporate the `1.0.1` C API
7. `1.0.2` of the C API is released
8. `1.0.4` of the Rust API is released to incorporate the `1.0.2` C API

From the version number alone, there is no way to know which C bugfixes are included in which Rust version. The intuitive rule, that all smaller patch versions are included doesn't hold because of 6 and 7.

Reply:
Thank you very much for your excellent feedback. It was incorporated in 03b6eb64bc24d172e1deb4c612ff0f83908634c9

> Semantic versioning allows to add a build metadata suffix. We could use that for the binding version, e.g. 1.2.3+rust1 is the first version of the Rust binding for 1.2.3, 1.2.3+rust2 is a second version that fixes some bugs only in the binding, etc.

We discussed this in #3308. Unfortunately the build metadata is unsuitable for most of the changes (more than rebuilding or docu) and there is no alternative in semver. It seems like semver simply did not consider downstream versions.

Reply:
I think that making the patch level binding-specific is the best solution. It is not ideal while we're on `0.9.x` but once `1.0` is out this will not matter as much.
I would also state what Elektra version the binding is currently compatible with in its Readme to make it explicit for users.